src/Resources/app/storefront/src/checkout/confirm-order.plugin.js

```diff
@@ -109,17 +112,27 @@ export default class ConfirmOrderPlugin extends Plugin {
 
     renderPaymentComponent(type) {
         if (type === 'oneclick') {
-            this.renderStoredPaymentMethodComponents()
+            this.renderStoredPaymentMethodComponents();
             return;
         }
+
+        let identifier = 'type';
+        // Filter payment method configs by brand in the case of giftcards
+        if (adyenCheckoutOptions.selectedPaymentMethodHandler.includes('giftcard')) {
+            identifier = 'brand';
+        }
+
         // Get the payment method object from paymentMethodsResponse
-        let filtered = $.grep(this.adyenCheckout.paymentMethodsResponse.paymentMethods, function(paymentMethod) {
-            return paymentMethod.type === type;
+        let paymentMethodConfigs = $.grep(this.adyenCheckout.paymentMethodsResponse.paymentMethods, function(paymentMethod) {
+            return paymentMethod[identifier] === type;
         });
-        if (filtered.length === 0) {
+        if (paymentMethodConfigs.length === 0) {
```

Comment:
can the `paymentMethodConfigs.length` be empty?
In case the method is not available in the payment methods response then wouldn't it be already filtered out on the backend?

Reply:
True, it is not likely to happen. This is just a paranoid check, just in case we mess up and the `this.adyenCheckout` is somehow missing the PM

Reply:
okay, it might be still useful for future debugging purposes then if such a thing happens or if the methods response changes